examples/state_machine/states.py

```diff
@@ -0,0 +1,361 @@
+        return PNTransition(
+            state=HandshakeFailedState,
+            context=self._context
+        )
+
+    def restored(self, event: events.SubscriptionRestoredEvent, context: PNContext) -> PNTransition:
+        return PNTransition(
+            state=ReceivingState,
+            context=self._context
+        )
+
+    def success(self, event: events.HandshakeReconnectSuccessEvent, context: PNContext) -> PNTransition:
+        return PNTransition(
+            state=ReceivingState,
+            context=self._context
+        )
+
+
+class HandshakeFailedState(PNState):
+    def __init__(self, context: PNContext) -> None:
+        super().__init__(context)
+        self._transitions = {
+            events.HandshakeReconnectRetryEvent.__name__: self.reconnect_retry,
+            events.SubscriptionChangedEvent.__name__: self.subscription_changed,
+            events.ReconnectEvent.__name__: self.reconnect,
+            events.SubscriptionRestoredEvent.__name__: self.subscription_restored,
+        }
+
+    def reconnect_retry(self, event: events.HandshakeReconnectRetryEvent, context: PNContext) -> PNTransition:
+        return PNTransition(state=HandshakeReconnectingState)
+
+    def subscription_changed(self, event: events.SubscriptionChangedEvent, context: PNContext) -> PNTransition:
+        return PNTransition(state=HandshakingState)
+
+    def reconnect(self, event: events.ReconnectEvent, context: PNContext) -> PNTransition:
+        return PNTransition(state=HandshakingState)
+
+    def subscription_restored(self, event: events.SubscriptionRestoredEvent, context: PNContext) -> PNTransition:
+        return PNTransition(
+            state=ReceivingState,
+            context=self._context
+        )
+
+
+class HandshakeStoppedState(PNState):
+    def __init__(self, context: PNContext) -> None:
+        super().__init__(context)
+        self._transitions = {
+            events.ReconnectEvent.__name__: self.reconnect
+        }
+
+    def reconnect(self, event: events.ReconnectEvent, context: PNContext) -> PNTransition:
+        return PNTransition(state=HandshakeReconnectingState)
+
+
+class ReceivingState(PNState):
+    def __init__(self, context: PNContext) -> None:
+        super().__init__(context)
+        self._transitions = {
+            events.SubscriptionChangedEvent.__name__: self.subscription_changed,
+            events.SubscriptionRestoredEvent.__name__: self.subscription_restored,
+            events.ReceiveSuccessEvent.__name__: self.receiving_success,
+            events.ReceiveFailureEvent.__name__: self.receiving_failure,
+            events.DisconnectEvent.__name__: self.disconnect,
+            events.ReconnectEvent.__name__: self.reconnect,
+        }
+
+    def on_enter(self, context: Union[None, PNContext]):
+        super().on_enter(context)
+        return effects.ReceiveEventsEffect(context.channels, context.groups)
```

Comment:
This effect should also receive `tt` and `region`

Reply:
Also we decided to rename it to `ReceiveMessages` to avoid conflicting with `PNEvent`